src/Microsoft.Health.Fhir.Shared.Api/Features/Formatters/FhirJsonOutputFormatter.cs

```diff
@@ -91,6 +96,12 @@ public override async Task WriteResponseBodyAsync(OutputFormatterWriteContext co
                         if (rawBundleEntryComponent is RawBundleEntryComponent)
                         {
                             rawBundleEntryComponent.Resource = ((RawBundleEntryComponent)rawBundleEntryComponent).ResourceElement.ToPoco<Resource>(_deserializer);
+                            if (hasElements)
```

Comment:
Nit: Could the if statements on lines 89 and 116 use hasElements instead of checking elementsSearchParameter?